library/lua/dfhack/buildings.lua

```diff
@@ -176,9 +176,9 @@ local building_inputs = {
             vector_id=df.job_item_vector_id.TRACTION_BENCH
         }
     },
-    [df.building_type.Slab] = { { item_type=df.item_type.SLAB } },
-    [df.building_type.NestBox] = { { has_tool_use=df.tool_uses.NEST_BOX, item_type=df.item_type.TOOL } },
-    [df.building_type.Hive] = { { has_tool_use=df.tool_uses.HIVE, item_type=df.item_type.TOOL } },
+    [df.building_type.Slab] = { { item_type=df.item_type.SLAB, vector_id=df.job_item_vector_id.SLAB } },
+    [df.building_type.NestBox] = { { has_tool_use=df.tool_uses.NEST_BOX, item_type=df.item_type.TOOL, vector_id=df.job_item_vector_id.TOOL } },
+    [df.building_type.Hive] = { { has_tool_use=df.tool_uses.HIVE, item_type=df.item_type.TOOL, vector_id=df.job_item_vector_id.TOOL } },
```

Comment:
These three `vector_id`s (well, two unique ones) are nil:
```
[lua]# ~df.job_item_vector_id.SLAB
nil
[lua]# ~df.job_item_vector_id.TOOL
nil
```
I see comments linking these to ANY_125 and ANY_124 (respectively) in df.item-vectors.xml, but those just point to `items.all`, which I think these were falling back to before anyway (but I'm having a hard time tracking down what actually uses the `vector_id` field...). At any rate, this doesn't hurt anything, since the value of `vector_id` was and is `nil`, but it could be confusing.

Reply:
Ah, I see I was assuming since the OTHER id exists, the vector id must as well. https://github.com/DFHack/df-structures/blob/master/df.item-vectors.xml#L669 mentions that SLAB (and similarly TOOL) is SKIPPED -- what does that mean (I don't see SKIPPED defined as a syntax item in SYNTAX.rst), any do you know why it would be skipped?

I'll revert these three back to undefined. Thank you for finding this!

Reply:
It just means "skipped", like how [EGG](https://github.com/DFHack/df-structures/blob/5756bf8599630891fc460b1509bd0660b71c0809/df.item-vectors.xml#L776) is [later](https://github.com/DFHack/df-structures/blob/5756bf8599630891fc460b1509bd0660b71c0809/df.item-vectors.xml#L744) than it would be if the order of `job_item_vector_id` matched `items_other_id`.

My understanding of `job_item_vector_id` is that it is a rough indicator of what types of items certain jobs require, given that it is only actually used in `job_item::vector_id`. It tends to correspond to `items_other_id` (the names deliberately match on our end when applicable, and sometimes the raw values happen to match too), but there are some entries interspersed that refer to ANY or different vectors altogether. Not sure why DF does it this way. I suspect that for "skipped" item types without an entry in `job_item_vector_id`, DF searches `world.items.all` instead - it would be slower, but any speedups from adding an entry would be relatively small. It's also possible that DF added a value at the end of `job_item_vector_id`, and we simply haven't noticed yet (since it's not used as an index the way `items_other_id` is).

Reply:
As a test, I created an "engrave memorial slab" job, and `job.job_items[0].vector_id` is 125 (job_item_vector_id::ANY_125). 
https://github.com/DFHack/df-structures/blob/5756bf8599630891fc460b1509bd0660b71c0809/df.item-vectors.xml#L772
It's possible that other job types also use `125`, so it can't always be resolved to `world.items.other.SLAB`. @quietust might be able to provide a better explanation.